### PR TITLE
feat(module-tools): add dts.abortOnError config

### PR DESCRIPTION
**Pull request by targeral**
## Description

1. add `dts.catchError` config and update docs.
2. fix temp tsconfig`s filename

## Types of changes

- [x] Docs change / Dependency upgrade
- [x] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/module-doc/docs/en/api/config/build-config.md [hunk 1]
```diff
@@ -183,17 +183,30 @@ To prevent excessive global replacement substitution, it is recommended that the
...
+The dts file generates the relevant configuration, by default it generates.
 
 - type: `false | Object`
-- default: `{}`
+- default:
 
-### tsconfigPath
+``` js
+{
+  catchError: false,
```
[the remark this answers is not in the document]
Comment (targeral) [reply packages/document/module-doc/docs/en/api/config/build-config.md:193]: OK

[the remark this answers is not in the document]
Comment (targeral) [reply packages/document/module-doc/docs/en/api/config/build-config.md]: done

File: packages/document/module-doc/docs/en/api/config/build-config.md [hunk 2]
```diff
@@ -183,17 +183,30 @@ To prevent excessive global replacement substitution, it is recommended that the
...
-- type: `string`
-- default: `. /tsconfig.json`
+Whether to allow the build to succeed in case of a type error. By default, this will cause the build to fail in case of a type error.
+
+:::warning
+When this configuration is turned on, there is no guarantee that the type files will be generated properly and accurately. In `buildType: 'bundle'` or Bundle build mode, the type file must not be generated.
+:::
+
+- 类型：`boolean`
+- 默认值：`false`
```
Comment (chenjiahan) [inline packages/document/module-doc/docs/en/api/config/build-config.md:209]: need translate

File: packages/document/module-doc/docs/zh/api/config/build-config.md [hunk 3]
```diff
@@ -236,17 +236,30 @@ export default defineConfig({
...
-- 类型： `string`
-- 默认值： `./tsconfig.json`
+在出现类型错误的时候，是否允许构建成功。**默认情况下，在出现类型错误的时候会导致构建失败**。
+
+:::warning
+当开启该配置后，无法保证类型文件正常生成以及准确。在 `buildType: 'bundle'` 或者 Bundle 构建模式下，类型文件一定不会生成。
+:::
+
+- 类型：`boolean`
+- 默认值：`false`
```
Comment (chenjiahan) [inline packages/document/module-doc/docs/zh/api/config/build-config.md:262]: default true

File: packages/document/module-doc/docs/zh/api/config/build-config.md [hunk 4]
```diff
@@ -236,17 +236,30 @@ export default defineConfig({
...
 
-TypeScript 配置文件的路径。
+### abortOnError
 
-- 类型： `string`
-- 默认值： `./tsconfig.json`
+在出现类型错误的时候，是否允许构建成功。**默认情况下，在出现类型错误的时候会导致构建失败**。
+
+:::warning
+当开启该配置后，无法保证类型文件正常生成以及准确。在 `buildType: 'bundle'` 或者 Bundle 构建模式下，类型文件一定不会生成。
```
Comment (chenjiahan) [inline packages/document/module-doc/docs/zh/api/config/build-config.md:258]:
```suggestion
当关闭该配置后，无法保证类型文件能正常生成，且不保证内容正确。在 `buildType: 'bundle'` 或者 Bundle 构建模式下，类型文件一定不会生成。
```

File: packages/document/module-doc/docs/zh/api/config/build-config.md [hunk 5]
```diff
@@ -236,17 +236,30 @@ export default defineConfig({
...
+类型文件生成的相关配置，默认情况会生成。
 
 - 类型： `false | Object`
-- 默认值： `{}`
+- 默认值：
 
-### tsconfigPath
+``` js
+{
+  abortOnError: false,
```
Comment (chenjiahan) [inline packages/document/module-doc/docs/zh/api/config/build-config.md:246]:
```suggestion
  abortOnError: true,
```

File: packages/solutions/module-tools/src/constants/build.ts [hunk 6]
```diff
@@ -11,6 +11,7 @@ export const defaultBuildConfig = Object.freeze<BaseBuildConfig>({
     only: false,
     distPath: './',
     tsconfigPath: './tsconfig.json',
+    abortOnError: false,
```
Comment (chenjiahan) [inline packages/solutions/module-tools/src/constants/build.ts:14]: true

File: .changeset/tender-bags-breathe.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+'@modern-js/module-tools': minor
+'@modern-js/module-tools-docs': minor
+---
+
+feat: add `dts.catchError` config
+feat: 添加 `dts.catchError` 配置
```
Comment (chenjiahan) [inline .changeset/tender-bags-breathe.md:7]: changeset should be updated~

**Pull request merged**